Corvus.Extensions.Samples/TraversalExtensionsSample.dib

```diff
@@ -0,0 +1,251 @@
+    return s;
+}
+
+Func<string, int, Task<string>> actionConsolePrint = ConsolePrint;
+
+#!csharp
+
+IEnumerable<string> list = new List<string> { "foo", "bar", "quux" };
+await list.ForEachAtIndexAsync<string>(actionConsolePrint);
+
+#!markdown
+
+### ForEachFailEnd()
+
+#!markdown
+
+Execute an action for each item in the enumerable. 
+
+If any operation fails, then the enumeration is continued to the end when an Aggregate Exception is thrown containing the exceptions thrown by any failed operations.
+
+#!csharp
+
+IEnumerable<int> list = new List<int> { 1, 2, 8, 0, 5 };
+
+static void Divide(int x)
+{
+    int result = 10/x;
+    Console.WriteLine(result);
+}
+
+Action<int> divideAction = Divide;
+
+list.ForEachFailEnd<int>(divideAction)
+
+#!markdown
+
+### ForEachFailEndAsync()
+
+#!markdown
+
+Execute an async action for each item in the enumerable.
+
+Returns a task which completes when the enumeration has completed.
+
+If any operation fails, then the enumeration is continued to the end when an Aggregate Exception is thrown containing the exceptions thrown by any failed operations.
+
+#!csharp
+
+IEnumerable<int> list = new List<int> { 1, 2, 8, 0, 5 };
+
+public async Task<int> DivideAsync(int x)
+{
+    int result = 10/x;
+    Console.WriteLine(result);
+    return result;
+}
+
+Func<int, Task<int>> divideAction = DivideAsync;
+
+await list.ForEachFailEndAsync<int>(divideAction)
+
+#!markdown
+
+### ForEachUntilFalse()
+
+#!markdown
+
+Execute an action for each item in the enumerable.
+
+Returns false if the enumeration returned early, otherwise true.
+
+#!csharp
+
+IEnumerable<int> list = new List<int> { 1, 2, 8, 0, 5 };
+
+static bool IsEqualToZero(int x)
+{
+    bool result = x.Equals(0);
+    return result;
+}
+
+Func<int, bool> equalsZeroAction = IsEqualToZero;
+
+list.ForEachUntilFalse<int>(equalsZeroAction)
+
+#!markdown
+
+### ForEachUntilFalseAsync()
+
+#!markdown
+
+Execute an async action for each item in the enumerable.
+
+A task which completes False if the enumeration returned early, otherwise true.
+
+#!csharp
+
+IEnumerable<int> list = new List<int> { 1, 2, 8, 0, 5 };
+
+public async Task<bool> IsEqualToZero(int x)
+{
+    bool result = x.Equals(0);
+    return result;
+}
+
+Func<int, Task<bool>> equalsZeroAction = IsEqualToZero;
+
+await list.ForEachUntilFalseAsync<int>(equalsZeroAction)
+
+#!markdown
+
+### ForEachUntilTrue()
+
+#!markdown
+
+Execute an action for each item in the enumerable.
+
+Returns true if the action terminated early, otherwise false.
+
+#!csharp
+
+IEnumerable<int> list = new List<int> { 1, 2, 8, 0, 5 };
+
+static bool IsEqualToZero(int x)
+{
+    bool result = x.Equals(0);
+    return result;
+}
+
+Func<int, bool> equalsZeroAction = IsEqualToZero;
+
+list.ForEachUntilTrue<int>(equalsZeroAction)
+
+#!markdown
+
+### ForEachUntilTrueAsync()
+
+#!markdown
+
+Execute an async action for each item in the enumerable.
+
+A task which completes True if the action terminated early, otherwise false.
```

Comment:
Probably better to start this sentence with something like "Returns a `Task` whose result is True if..."